zipline/assets/assets.py

```diff
@@ -506,14 +627,17 @@ def map_identifier_index_to_sids(self, index, as_of_date):
         # Return a list of the sids of the found assets
         return [asset.sid for asset in matches]
 
-    def insert_metadata(self, identifier, **kwargs):
+    def _insert_metadata(self, identifier, **kwargs):
         """
         Inserts the given metadata kwargs to the entry for the given
         identifier. Matching fields in the existing entry will be overwritten.
         :param identifier: The identifier for which to insert metadata
         :param kwargs: The keyed metadata to insert
         """
-        entry = self.metadata_cache.get(identifier, {})
+        if identifier in self.metadata_cache:
+            # Multiple pass insertion no longer supported.
+            return
```

Comment:
We should, but that would currently be incompatible with the tests in test_algorithm, which insert multiple empty values for sid 133 on data source creation.

We could and should do another round of clean up there. I'll add our desire to raise an Exception in the comment here.